### ISSUE TITLE
Three small improvements in File > Import from Library

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#7205]
@shadrackkibet these are pretty trivial - well 2 are, and the 3rd is optional, and only if you agree?
1. The menu item is now Import from Library.  So please change the title in the dialogue from `Open Dataset from Library` to` Import from Library`. 
2. There is a second Help button on the right of the dialogue, just below the list of files.  Change this from `Help` to `R Help`, so it is distinguished from the main Help button.  Alternatively it could be` Info`, but I think `R Help` will be fine.
3. I find I often do a double-click on a data set and expect it to work.  Now that is unusual in that we don't have any other dialogues that use double-click to implement Ok.  We do have double-click in the data selector to copy a selected variable to the receiver, so it is consistent with that.  I would certainly find it convenient.    Danny is Ok with the idea - at least not against it!

We will soon have a video on this dialogue.

**Comment by shadrackkibet** [comment]
I agree with all these and can be done by an intern. @Gioche6 do you want to give it a try?

**Comment by Gioche6** [comment]
@shadrackkibet , I'll take this up. Thanks

**Comment by rdstern** [comment]
Thanks @Gioche6 .  I have a further query, in addition to these 3 tasks - of course.
My example is the library janeaustenr, which has 6 data sets.
In the dialogue we can currently only import one file at a time.  Could we implement the facility to import multiple files.  So implementing <ctrl> click and <shift> click to be able to select multiple files, and then import them into separate data frames.
If you need help, then @N-thony has just implemented something similar for File > Import.

**Comment by N-thony** [comment]
> Thanks @Gioche6 . I have a further query, in addition to these 3 tasks - of course. My example is the library janeaustenr, which has 6 data sets. In the dialogue we can currently only import one file at a time. Could we implement the facility to import multiple files. So implementing click and click to be able to select multiple files, and then import them into separate data frames. If you need help, then @N-thony has just implemented something similar for File > Import.

@Gioche6 I think for the multiple selection, it may be simple to implement than I thought, I have tried to run
`data_book$import_data(data_tables=list(attenu=attenu, attitude = attitude, airquality = airquality))` and you can see the result in the screenshot below, and it loads correctly the three datasets with their corresponding names.
So, have a look at the VB code, and try it, let me know if you get stuck. You will need also to activate `multiselect` to `True` in the property of the `ListView` i.e `lstCollection` that contains dataset names on the dialogue.

![image](https://user-images.githubusercontent.com/68591383/155684384-9a875525-d0a9-41d2-97aa-26a4a9a805fc.png)

**Comment by N-thony** [comment]
@derekagorhom can you this too?

**Comment by derekagorhom** [comment]
yes please i would like to work on this

**Comment by N-thony** [comment]
> yes please i would like to work on this

 @derekagorhom I was helping Dennis on this and made some progress, have a look of the change in #7244 so far, and then we will continue.

**Comment by N-thony** [comment]
> > yes please i would like to work on this
> 
> @derekagorhom I was helping Dennis on this and made some progress, have a look of the change in #7244 so far, and then we will continue.

@derekagorhom how is it going with this?

**Comment by derekagorhom** [comment]
Closed #7205

**Comment by rdstern** [comment]
@N-thony I found this issue by accident, but have two questions that relate to it.
a) The double-click now works fine on the Import from Library.  But that is almost the only occasion when we use the double-click.
I wonder whether @derekagorhom could also implement double-click on the recent files?  Well I guess he can - but do you agree this would be a welcome improvement?  Currently we always get the importing dialog, and @rachelkg was wondering which options we use - and I have never used any of them in 8 years!  So, maybe the availability of the double-click would simplify the re-importing of recent data, and so simplify R-Instat a bit?

But I also report here that starting R-Instat and then going to Import from Library often takes a little time and then closes R-Instat.  I simply click on the R-Instat icon in the task bar, so it is no big deal - and it is inconsistent, but I wonder why?

**Comment by Rockjunior** [comment]
@N-thony @Ag-Derek @rdstern @berylwaswa 
I have gone through the issue and traced the related PRs, and from my checks, the concerns raised here appear to have already been implemented, either directly or through subsequent improvements. On my side, the current version of R-Instat seems to be working correctly, especially considering this issue was last reviewed in the first quarter of 2024.

Could someone else kindly verify this as well so we can confirm whether the issue can now be closed?

**Comment by berylwaswa** [comment]
@Rockjunior I have gone through the issues raised, and most have been implemented. Except we still do not have a double click for the recent files, when I double click it just opens the Import from file dialog, so maybe get that done and we can close it. Is this still something you'd like done @rdstern ?

**Comment by Rockjunior** [comment]
@berylwaswa yes please, looking into it

**Comment by Rockjunior** [comment]
I attempted to implement direct opening of recent datasets by bypassing the import dialog and reusing the existing import logic through an `ImportFileDirect` helper integrated with `clsRecentFiles`. However, the approach introduced multiple compile and integration issues, and I was not able to get it working reliably.

The previous PR therefore does not resolve the issue as intended, so I’ll be closing it for now. Sharing this update for visibility since I did spend some time exploring the approach, but it likely needs a cleaner implementation or refactor to work properly.
